src/ApparentHorizons/StrahlkorperGr.cpp

```diff
@@ -85,6 +85,42 @@ Scalar<DataVector> expansion(
   return expansion;
 }
 
+template <typename Frame>
+tnsr::ii<DataVector, 3, Frame> extrinsic_curvature(
+    const tnsr::ii<DataVector, 3, Frame>& grad_normal,
+    const tnsr::i<DataVector, 3, Frame>& unit_normal_one_form,
+    const tnsr::II<DataVector, 3, Frame>& upper_spatial_metric) noexcept {
+  // Form projector
+  tnsr::Ij<DataVector, 3, Frame> projector(get<0>(unit_normal_one_form).size(),
+                                           0.0);
+  const auto& unit_normal_vector =
+      raise_or_lower_index(unit_normal_one_form, upper_spatial_metric);
+  for (size_t i = 0; i < 3; ++i) {
+    projector.get(i, i) += 1.0;
+    for (size_t j = 0; j < 3; ++j) {
+      projector.get(i, j) -=
+          unit_normal_vector.get(i) * unit_normal_one_form.get(j);
+    }
+  }
+
```

Comment:
It will be more efficient to not compute the projector.  Instead
0) Allocate K_ij = grad_normal_ij
1) Compute say temp = n^k n^l  grad_normal_kl
2) Then compute K_ij  +=  temp n_i n_j
3) K_ij -= 2*grad_normal_l(i n_j) n^l